app/subjects/index.cjsx

```diff
@@ -0,0 +1,81 @@
+React = require 'react'
+apiClient = require '../api/client'
+authClient = require '../api/auth'
+talkClient = require '../api/talk'
+getSubjectLocation = require '../lib/get-subject-location'
+FavoritesButton = require '../collections/favorites-button'
+ChangeListener = require '../components/change-listener'
+PromiseRenderer = require '../components/promise-renderer'
+NewDiscussionForm = require '../talk/discussion-new-form'
+{Navigation} = require 'react-router'
+
+module?.exports = React.createClass
+  displayName: 'Subject'
+  mixins: [Navigation]
+
+  getInitialState: ->
+    subject: null
+
+  componentWillMount: ->
+    @setSubject()
```

Comment:
This should also get called from `componentWillReceiveProps` (`if nextProps.params?.id isn't @props.params?.id`).